s3.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"net/http"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	awshttp "github.com/aws/aws-sdk-go-v2/aws/transport/http"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	log "github.com/sirupsen/logrus"
+)
+
+type MetadataBackend struct {
+	Client *s3.Client
+	Bucket string
+}
+
+func connect_to_s3(mConf MetadataS3Config) *MetadataBackend {
+	httpClient := awshttp.NewBuildableClient().WithTransportOptions(func(tr *http.Transport) {
+		if tr.TLSClientConfig == nil {
+			tr.TLSClientConfig = &tls.Config{}
+		}
+		tr.TLSClientConfig.MinVersion = tls.VersionTLS13
+	})
+
+	cfg, err := config.LoadDefaultConfig(context.TODO(),
+		config.WithRegion(mConf.Region),
+		config.WithHTTPClient(httpClient),
+		config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(mConf.AccessKey, mConf.SecretKey, "")),
+		config.WithEndpointResolverWithOptions(aws.EndpointResolverWithOptionsFunc(
+			func(service, region string, options ...interface{}) (aws.Endpoint, error) {
+				return aws.Endpoint{URL: mConf.URL}, nil
+			})),
+	)
+
+	if err != nil {
+		log.Fatal("Could not connect to metadata S3 bucket: ", err)
```

Comment:
```suggestion
		log.Fatalf("Error while setting up S3 config: %v", err)
```

Probably not a problem here, but generally we want to use a format rather than pass objects directly to the formatter. Also adjust message because I think the connection hasn't been tried yet here?

Reply:
changes made for handling config errors and s3 bucket connection